src/main.rs

```diff
@@ -161,36 +184,43 @@ async fn apply(
         .ok_or(Error::MalformedProviderId)?
         .1;
 
-    let instance_description = describe_instance(&ec2_client, instance_id.to_owned()).await?;
+    let eni_id = match get_eni_id_from_annotation(&pod) {
+        Some(eni_id) => eni_id,
+        None => {
+            let instance_description =
+                describe_instance(&ec2_client, instance_id.to_owned()).await?;
+
+            instance_description
+                .reservations
+                .as_ref()
+                .ok_or(Error::MissingReservations)?[0]
+                .instances
+                .as_ref()
+                .ok_or(Error::MissingInstances)?[0]
```

Comment:
maybe check here that it's only attached to one instance? I imagine it would never be attached to more, but who knows what fun wedged states you can get into with AWS

Reply:
Also, at this size of `.ok_or` chaining, I'd suggest using some `if let` match instead.

Reply:
Do you have an example of how to do the `if let` so it handles this?

I'm specifically asking it to describe instances with a particular instance ID, so it should never have more than one.

All this code is just moved inside the `None` match branch. It hasn't changed at all. I'd love to improve this code, but I think that can go in a separate PR.